src/gui/render/graphics_shaders/program_handler.cpp

```diff
@@ -235,9 +204,63 @@ Program::get_status_string() const {
 
 void
 Program::attach_uniforms() {
-    for (const auto& uniform_names : found_uniforms_) {
-        const auto& name = uniform_names.first;
-        uniforms_[name] = new_uniform(name);
+    GLint count;
+    const GLsizei buf_size = 64;
+    GLchar name[buf_size];
+    GLsizei length;
+    GLint size;
+    GLenum type;
+
+    glGetProgramiv(program_ID_, GL_ACTIVE_UNIFORMS, &count);
+
+    for (GLuint uid = 0; uid < static_cast<GLuint>(count); uid++) {
+        glGetActiveUniform(program_ID_, uid, buf_size, &length, &size, &type, name);
+
+        std::string str_name(name);
+
+        gpu_data::GPUDataType enum_type =
+            static_cast<gpu_data::GPUDataType>(type); // this might fail
```

Comment:
At the end of your GPUDataType enum add a NUM_TYPES member, and then you can check `0 < type && type < GPUDataType::NUM_TYPES`